WordPress/twentytwentyfour
# Change post pattern on `home.html`, update template slugs

**Description**

This PR fixes the issues around template names and query loops not inheriting the default loop, raised in #705 and #404. It replaces the original `posts-grid-2-col` pattern with the `posts-3-col` pattern, wrapped in a group block to align with the rest of the layout and the original h2 headline. This fixes the issues raised in #705 with ignoring the main query and breaking the navigation.

Renaming the `home.html` template file to `front-page.html` fixes the issue when a separate blog page is set, giving the users full control over the theme with the settings they make in Settings -> Reading. This was raised in #404.

In addition, I've renamed the slugs of the template and page to the full file names, so they're the same as the other templates and patterns and are easier recognizable.

I'm aware this "breaks" aka reverts the original design, but I have to agree with @felixarntz that we should aim to not break the core functionality of the main query. I'm currently not aware what the capabilities of the query loop block in terms of design are and if there can be a different design for the query, but for now we should stick to this. If users want to go back to the original design with the three posts, they can totally do so and replace the main query, but then it's their active decision.

I'd be interested in getting feedback from @MaggieCabrera, @beafialho, @richtabor and anyone in @WordPress/block-themers.

**Screenshots**
The screenshot shows my test environment with the recent changes, including how 10 posts with images and an extra sticky post would look like on the main page, with the default WordPress settings.


<img width="1280" alt="CleanShot 2023-10-27 at 21 33 18@2x" src="https://github.com/WordPress/twentytwentyfour/assets/7585600/eab4186c-8c10-491a-aa6f-01d46632b702">


**Testing Instructions**

1. Check out this PR
2. Go to Appearance -> Site Editor
3. Click on Templates
4. Click on "Manage all templates" at the bottom
5. Make sure to revert any customizations if necessary
6. Check out the Front Page template
7. Open the frontend
8. Test the navigation/pagination
